k2/python/k2/fsa.py

```diff
@@ -249,12 +249,17 @@ def _invalidate_cache_(self, scores_only: bool = True) -> None:
             pattern = re.compile('score')
             to_remove = []
 
-            for key in self.__dict__['_cache']:
```

Comment:
I think self.__dict__['cache'] was faster, because we avoid going through our __getitem__() code.

Reply:
ok, reverted.